Add Odroid H4 profile

3mkusiak opened the pull request:
Add profile for Odroid H4.

dancesWithMachines committed:
dts: dts-gen-profile: Pass heads for configs

Enforce passing refs for dts-configs in DTG001 in dts-gen-profile.robot.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
dts/profiles: Add odroid h4 update profile

Add UEFI update profile for Hardkernel Odroid H4 plus.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

m-iwanicki commented on dts/dts-gen-profile.robot:
```diff
@@ -168,6 +168,7 @@ Prepare DTS For Profile Generation
     Execute Command In Terminal    mkdir -p /tmp/bin
     Execute Command In Terminal    echo '#!/bin/bash' >/tmp/bin/reboot
     Execute Command In Terminal    chmod +x /tmp/bin/reboot
+    Execute Command In Terminal    export DTS_CONFIG_REF="${dts_config_ref}"
```
```suggestion
    Execute Command In Terminal    export DTS_CONFIG_REF="${DTS_CONFIG_REF}"
```

global variables should be capitalized. Install pre-commit as it should've told you that.

3mkusiak replied:
[Done](https://github.com/Dasharo/open-source-firmware-validation/pull/1070/commits/d0e5596c7948a670e21f18c1a9ec84547c9041af)

m-iwanicki commented on dts/profiles/odroid-h4-plus UEFI Update - DPP.profile:
```diff
@@ -0,0 +1,40 @@
+flashrom -p internal -N --ifd -i fd -w /tmp/biosupdate 0
+reboot  0
+dmidecode  0
+dmidecode  0
```
Remove last dmidecode, that's a bug that happens for some workflows and some platforms when generating profile automatically.

3mkusiak replied:
[Done](https://github.com/Dasharo/open-source-firmware-validation/pull/1070/commits/1ec797d7531cf5816213a677836236e0b2fd3ebd)

dancesWithMachines committed:
dts/profiles: odroid h4 update: remove dmidecode

Remove extra dmidecode from the profile for odroid h4 plus for UEFI
update. It is a known bug.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
dts: dts-gen-profile: Fix case for config var

Fix case for exporting dts_config_ref heads in dts-gen-profile.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

dancesWithMachines committed:
configs: odroid h4: add update to workflow profs

Add "UEFI Update" to DTS_TEST_WORKFLOW_PROFILES for Odroid H4 plus.

Signed-off-by: Mateusz Kusiak <mateusz.kusiak@3mdeb.com>

DaniilKl commented:
Rebased on latest develop.
